server/src/test/java/org/elasticsearch/snapshots/mockstore/MockEventuallyConsistentRepositoryTests.java

```diff
@@ -0,0 +1,126 @@
+                final byte[] readBytes = new byte[lengthWritten + 1];
+                final int lengthSeen = in.read(readBytes);
+                assertThat(lengthSeen, equalTo(lengthWritten));
+                assertArrayEquals(blobData, Arrays.copyOf(readBytes, lengthWritten));
+            }
+        }
+    }
+
+    public void testReadAfterWriteAfterReadThrows() throws IOException {
+        MockEventuallyConsistentRepository.Context blobStoreContext = new MockEventuallyConsistentRepository.Context();
+        try (BlobStoreRepository repository = new MockEventuallyConsistentRepository(
+            new RepositoryMetaData("testRepo", "mockEventuallyConsistent", Settings.EMPTY), environment,
+            xContentRegistry(), mock(ThreadPool.class), blobStoreContext)) {
+            repository.start();
+            final BlobContainer blobContainer = repository.blobStore().blobContainer(repository.basePath());
+            final String blobName = randomAlphaOfLength(10);
+            final int lengthWritten = randomIntBetween(1, 100);
+            final byte[] blobData = randomByteArrayOfLength(lengthWritten);
+            assertMissing(blobContainer, blobName);
+            blobContainer.writeBlob(blobName, new ByteArrayInputStream(blobData), lengthWritten, true);
+            assertThrowsOnInconsistentRead(blobContainer, blobName);
+        }
+    }
+
+    public void testReadAfterDeleteAfterWriteThrows() throws IOException {
+        MockEventuallyConsistentRepository.Context blobStoreContext = new MockEventuallyConsistentRepository.Context();
+        try (BlobStoreRepository repository = new MockEventuallyConsistentRepository(
+            new RepositoryMetaData("testRepo", "mockEventuallyConsistent", Settings.EMPTY), environment,
+            xContentRegistry(), mock(ThreadPool.class), blobStoreContext)) {
+            repository.start();
+            final BlobContainer blobContainer = repository.blobStore().blobContainer(repository.basePath());
+            final String blobName = randomAlphaOfLength(10);
+            final int lengthWritten = randomIntBetween(1, 100);
+            final byte[] blobData = randomByteArrayOfLength(lengthWritten);
+            blobContainer.writeBlob(blobName, new ByteArrayInputStream(blobData), lengthWritten, true);
+            blobContainer.deleteBlob(blobName);
+            assertThrowsOnInconsistentRead(blobContainer, blobName);
+            blobStoreContext.forceConsistent();
+            assertMissing(blobContainer, blobName);
+        }
+    }
+
+    private static void assertThrowsOnInconsistentRead(BlobContainer blobContainer, String blobName) throws IOException {
+        try (InputStream in = blobContainer.readBlob(blobName)) {
+            fail("Inconsistent read should throw");
+        } catch (AssertionError assertionError) {
+            assertThat(assertionError.getMessage(), equalTo("Inconsistent read on [" + blobName + ']'));
```

Comment:
use expectThrows?

Reply:
Done in 58a9d4234b6fe151743bb3bbf842b5ef704e6ba9